### PR TITLE
chore: Update release settings for JReleaser 0.9

### PR DESCRIPTION
**Pull request by aalmiray**
Update the JReleaser configuration ad GH workflows with latest changes required by v0.9

### CODE REVIEW

**Comment by aalmiray** [comment]
Converting to draft as JReleaser v0.9 is unreleased and JBang prefers using released versions.

File: jreleaser.yml [hunk 1]
```diff
@@ -66,6 +67,7 @@ files:
 
 distributions:
   jbang:
+    executableExtension: cmd
```
Comment (maxandersen) [inline jreleaser.yml:70]: what does this mean?

Comment (aalmiray) [reply jreleaser.yml:70]: By default JReleaser uses .bat as the file extension for executables/launchers on Windows. JBang uses .cmd. The leading . is not necessary in the config

**Comment by aalmiray** [comment]
@maxandersen feel free to merge any time now that JReleaser v0.9.0 is out.

**Comment by maxandersen** [comment]
merged after bumping it to actually use 0.9 - thanks!

**Pull request merged**